projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func main() {
+	for {
+		weather, err := getWeatherWithRetries(3) // Retry up to 3 times
```

Comment:
I'd generally recommend centralising your sleeping logic in one place, and your retrying logic in one place. It looks like here you have a `for` loop inside which you call a function which does its own retries - I'd recommend only having one piece of code doing retries (and if you need to pass information like "how many retries are remaining" or "how many retries have been attempted" around to the place that decides to do the retries, to do so).

Similarly with the sleeping, it looks like you have three separate sleeps - one in this `for` loop, one in the retry function, and one in the `getWeather` function after the `Retry-After` header is parsed. Having these separate makes it hard to reliably sleep for the right amount of time (e.g. if `Retry-After` says sleep for 3 seconds, you will actually sleep for at least 4 because of the extra sleep in `getWeatherWithRetries`).